src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
> Where there is an Option could you add a null case to the enum and ensure it gets serialised as 0 or null or whatever the protocol wants?

The enum itself is not defined with a "null" case so adding one to the enum would just be more confusing it kind of smells like a null pointer.